test/quantization/fx/test_model_report_fx.py

```diff
@@ -1179,3 +1200,118 @@ def test_input_weight_equalization_determine_points(self):
                     else:
                         # if it's not a supported type, it shouldn't have observer attached
                         self.assertTrue(not hasattr(module, InputWeightEqualizationDetector.DEFAULT_PRE_OBSERVER_NAME))
+
+    @skipIfNoFBGEMM
+    def test_input_weight_equalization_report_gen(self):
+        # use fbgemm and create our model instance
+        # then create model report instance with detector
+        with override_quantized_engine('fbgemm'):
+
+            test_input_weight_detector = InputWeightEqualizationDetector(0.4)
+            detector_set = set([test_input_weight_detector])
+            # prepare the model for callibration
+            prepared_for_callibrate_model, model_report = self._get_prepped_for_calibration_model(
+                self.TwoBlockComplexNet(), detector_set
+            )
+
+            # now we actually callibrate the model
+            example_input = torch.arange(27).reshape((1, 3, 3, 3))
+            example_input = example_input.to(torch.float)
+
+            prepared_for_callibrate_model(example_input)
+
+            # now get the report by running it through ModelReport instance
+            generated_report = model_report.generate_model_report(prepared_for_callibrate_model, True)
+
+            # check that sizes are appropriate only 1 detector
+            self.assertEqual(len(generated_report), 1)
+
+            # get the specific report for input weight equalization
+            input_weight_str, input_weight_dict = generated_report[test_input_weight_detector.get_detector_name()]
+
+            # we should have 5 layers looked at since 5 conv / linear layers
+            self.assertEqual(len(input_weight_dict), 5)
+
+            # we can validate that the max and min values of the detector were recorded properly for the first one
+            # this is because no data has been processed yet, so it should be values from original input
+
+            example_input = example_input.reshape((3, 3, 3))  # reshape input
+            for module_fqn in input_weight_dict:
+                # look for the first linear
+                if "block1.linear" in module_fqn:
+                    block_1_lin_recs = input_weight_dict[module_fqn]
+                    # get input range info and the channel axis
+                    input_range_info = block_1_lin_recs[InputWeightEqualizationDetector.INPUT_INFO_KEY]
```

Comment:
will users just be looking at the report or will they look at these informations as well? I think we should focus on testing what is used by user in the end

Reply:
This information is also passed as a dict object to the user, so they can perform their own analysis afterwards. Because the information is user facing, I thought it would be worth it to test.